packages/react/src/publication/useSearchPublications.ts

```diff
@@ -0,0 +1,39 @@
+import {
+  CommentFragment,
+  CustomFiltersTypes,
+  PostFragment,
+  useSearchPublicationsQuery,
+} from '@lens-protocol/api-bindings';
+
+import { PaginatedArgs, PaginatedReadResult, usePaginatedReadResult } from '../helpers';
+import { useSharedDependencies } from '../shared';
+import { DEFAULT_PAGINATED_QUERY_LIMIT } from '../utils';
+
+type UseSearchPublicationsArgs = PaginatedArgs<{
+  query: string;
+  limit?: number;
+  observerId?: string;
+  customFilters?: Array<CustomFiltersTypes>;
+}>;
+
+export function useSearchPublications({
+  query,
+  limit = DEFAULT_PAGINATED_QUERY_LIMIT,
+  observerId,
+  customFilters,
```

Comment:
What's the implication of exposing `customFilters`? If we know them and it's a finite set let's formalize them with clear types, if not, better hold for now until we clarify what use cases we need to support.

Let's make sure we don't expose too low level details of the API which would be detrimental to the purpose of this SDK (simplify interactions with the Lens Protocol).

Reply:
At the moment `customFilters` is an enum (this may or not change in the future) so we know the values and they're defined in the generated codegen file. With this PR we are exporting it in the `react` package.
For now users will be able to use the only defined filter `CustomFiltersTypes.Gardeners` which filter some spam. I think this one is pretty useful for any ui to delete spam from their explore/search pages.

Reply:
I opted for deleting the `customFilters` in my last commit, i don't want to stale this PR anymore, we can discuss in the future if it's worth adding.